[ANNOUNCEMENT] Status of Xcode 10 support

A few issues have started to come in regarding the state of Cordova projects on Xcode 10. This is a rough summary of the situation:

Xcode 10 uses a new build system by default (previously available on an opt-in basis in Xcode 9). The cordova-ios project structure is not compatible with this new build system and results in failures.
Officially, we do not claim to support Xcode 10.

Currently the best workaround is to opt-out of the new build system:

* If you're building on the command-line, you can specify `--buildFlag="-UseModernBuildSystem=0"`:
   ```
   # Cordova CLI
   cordova run ios --buildFlag='-UseModernBuildSystem=0'
   cordova build ios --buildFlag='-UseModernBuildSystem=0'
   
   # Ionic CLI
   ionic cordova run ios -- --buildFlag="-UseModernBuildSystem=0"
   ionic cordova build ios -- --buildFlag="-UseModernBuildSystem=0"
   ```

* If you're building with a build.json config file, you can add the following under the iOS release or debug config:
   ```json
   "buildFlag": [
     "-UseModernBuildSystem=0"
   ]
   ```

* If you are opening the project in the Xcode IDE, you need to change the build system in Workspace Settings to "Legacy Build System"

  <details><summary>Xcode example</summary>
  <img src="https://user-images.githubusercontent.com/781366/45737593-f7e30780-bbf6-11e8-86f8-207dd596ad36.png" alt="Workspace Settings is under the File menu">
  <img src="https://user-images.githubusercontent.com/781366/45737603-fc0f2500-bbf6-11e8-84c9-8dcb17e7c691.png" alt="Change the Build System option to Legacy Build System">

  Thanks [**kitolog**](https://github.com/apache/cordova-ios/issues/412#issuecomment-422689481) for the screenshots!
  </details>

We're going to investigate what's required to make cordova-ios compatible with the new build system, and hope to include that in the next major version (cordova-ios@5.0.0).

@jeffreyramia with the help of your answer I am able to build an app but whenever I am trying to emulate it gives me an error as "platforms/ios/build/emulator/demo APP.app/Info.plist file not found.
[ERROR] An error occurred while running subprocess cordova."

> > @nagthgr8 you should use:
> > ```shell
> > ionic cordova build ios --prod --release -- --buildFlag="-UseModernBuildSystem=0"
> > ```
> 
> @gabriellacerda yes I am using the same it fails with below compiler error
> ionic cordova build ios --release --prod – --buildFlag="-UseModernBuildSystem=0"
> 
> Below compiler error I am getting while executing above command in MAC with xCode 10
> 
> 74-80-250-211:ios user913708$ [ERROR] An error occurred while running subprocess ionic-app-scripts.
> 
> -bash: [ERROR]: command not found
> 
> 74-80-250-211:ios user913708$
> 
> 74-80-250-211:ios user913708$ ionic-app-scripts build --prod --target cordova --platform ios exited
> 
> ```
> with exit code 1.
> 
> 
> 
> Re-running this command with the --verbose flag may provide more 
> 
> information.[12:36:57]  ionic-app-scripts 3.2.0 
> ```
> [12:36:57] build prod started …
> 
> [12:36:57] ionic-app-script task: “build”
> 
> [12:36:57] Error: tsconfig: Cannot read file ‘/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json’:
> 
> ```
>     ENOENT: no such file or directory, open 
> 
>     '/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json'. 
> ```
> Error: tsconfig: Cannot read file ‘/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json’: ENOENT: no such file or directory, open ‘/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json’.
> 
> at new BuildError (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
> 
> at getTsConfig (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:313:15)
> 
> at Object. (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:298:46)
> 
> at step (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:32:23)
> 
> at Object.next (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:13:53)
> 
> at /Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:7:71
> 
> at new Promise ()
> 
> at __awaiter (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:3:12)
> 
> at Object.getTsConfigAsync (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:295:12)
> 
> at Object.validateRequiredFilesExist (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/build/util.js:89:21)
> 74-80-250-211:ios user913708$ with exit code 1

Ok, it seems the compiler is not finding your ``tsconfig.json`` config file, try removing the platform and adding it again:
````sh
ionic cordova platform remove ios; ionic cordova platform add ios
````
Then try building again...

> > > @nagthgr8 you should use:
> > > ```shell
> > > ionic cordova build ios --prod --release -- --buildFlag="-UseModernBuildSystem=0"
> > > ```
> > 
> > 
> > @gabriellacerda yes I am using the same it fails with below compiler error
> > ionic cordova build ios --release --prod – --buildFlag="-UseModernBuildSystem=0"
> > Below compiler error I am getting while executing above command in MAC with xCode 10
> > 74-80-250-211:ios user913708$ [ERROR] An error occurred while running subprocess ionic-app-scripts.
> > -bash: [ERROR]: command not found
> > 74-80-250-211:ios user913708$
> > 74-80-250-211:ios user913708$ ionic-app-scripts build --prod --target cordova --platform ios exited
> > ```
> > with exit code 1.
> > 
> > 
> > 
> > Re-running this command with the --verbose flag may provide more 
> > 
> > information.[12:36:57]  ionic-app-scripts 3.2.0 
> > ```
> > [12:36:57] build prod started …
> > [12:36:57] ionic-app-script task: “build”
> > [12:36:57] Error: tsconfig: Cannot read file ‘/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json’:
> > ```
> >     ENOENT: no such file or directory, open 
> > 
> >     '/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json'. 
> > ```
> > Error: tsconfig: Cannot read file ‘/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json’: ENOENT: no such file or directory, open ‘/Users/user913708/vedichoroo/vedichoroo/platforms/ios/tsconfig.json’.
> > at new BuildError (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
> > at getTsConfig (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:313:15)
> > at Object. (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:298:46)
> > at step (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:32:23)
> > at Object.next (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:13:53)
> > at /Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:7:71
> > at new Promise ()
> > at __awaiter (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:3:12)
> > at Object.getTsConfigAsync (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/transpile.js:295:12)
> > at Object.validateRequiredFilesExist (/Users/user913708/vedichoroo/vedichoroo/node_modules/@ionic/app-scripts/dist/build/util.js:89:21)
> > 74-80-250-211:ios user913708$ with exit code 1
> 
> Ok, it seems the compiler is not finding your `tsconfig.json` config file, try removing the platform and adding it again:
> 
> ```shell
> ionic cordova platform remove ios; ionic cordova platform add ios
> ```
> 
> Then try building again...

@gabriellacerda I followed your instructions while add command is executed I get below response
>ionic cordova platform add ios
ENDENT:  no such file or directory, uv_cwd()

--- a/bin/templates/scripts/cordova/lib/copy-www-build-step.js
+++ b/bin/templates/scripts/cordova/lib/copy-www-build-step.js
@@ -51,9 +51,9 @@ try {
 // Code signing files must be removed or else there are
 // resource signing errors.
 shell.rm('-rf', dstWwwDir);
-shell.rm('-rf', path.join(dstDir, '_CodeSignature'));
-shell.rm('-rf', path.join(dstDir, 'PkgInfo'));
-shell.rm('-rf', path.join(dstDir, 'embedded.mobileprovision'));
+// shell.rm('-rf', path.join(dstDir, '_CodeSignature'));
+// shell.rm('-rf', path.join(dstDir, 'PkgInfo'));
+// shell.rm('-rf', path.join(dstDir, 'embedded.mobileprovision'));
 
 // Copy www dir recursively
 var code;

--- a/bin/templates/scripts/cordova/lib/copy-www-build-step.js
+++ b/bin/templates/scripts/cordova/lib/copy-www-build-step.js
@@ -48,12 +48,7 @@ try {
     process.exit(2);
 }
 
-// Code signing files must be removed or else there are
-// resource signing errors.
 shell.rm('-rf', dstWwwDir);
-// shell.rm('-rf', path.join(dstDir, '_CodeSignature'));
-// shell.rm('-rf', path.join(dstDir, 'PkgInfo'));
-// shell.rm('-rf', path.join(dstDir, 'embedded.mobileprovision'));
 
 // Copy www dir recursively
 var code;

